contracts/FlowIDTableStaking.cdc

```diff
@@ -886,26 +902,91 @@ pub contract FlowIDTableStaking {
             }
             emit EpochTotalRewardsPaid(total: totalRewards, fromFees: fromFees, minted: mintedRewards, feesBurned: rewardsVault.balance)
 
+            // Clear the non-operational node list so it doesn't persist to the next rewards payment
+            let emptyNodeList: {String: UFix64} = {}
+            self.setNonOperationalNodesList(emptyNodeList)
+
             // Destroy the remaining fees, even if there are some left
             destroy rewardsVault
         }
 
+        /// Calculates rewards for all the staked node operators and delegators
         pub fun calculateRewards(): [RewardsBreakdown] {
             let allNodeIDs = FlowIDTableStaking.getNodeIDs()
 
+            // Get the sum of all tokens staked
             var totalStaked = FlowIDTableStaking.getTotalStaked()
             if totalStaked == 0.0 {
                 return []
             }
+            // Calculate the scale to be multiplied by number of tokens staked per node
             var totalRewardScale = FlowIDTableStaking.epochTokenPayout / totalStaked
 
             var rewardsBreakdownArray: [FlowIDTableStaking.RewardsBreakdown] = []
 
+            // The total rewards that are withheld from the non-operational nodes
+            var sumRewardsWithheld = 0.0
+
+            // Iterate through all the non-operational nodes and calculate
+            // their rewards that will be withheld
+            let nonOperationalNodes = FlowIDTableStaking.getNonOperationalNodesList()
+            for nodeID in nonOperationalNodes.keys {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+
+                // Each node's rewards can be decreased by a different percentage
+                // Its delegator's rewards are also decreased by the same percentage
+                let rewardDecreasePercentage = nonOperationalNodes[nodeID]!
+
+                // Calculate the normal reward amount, then the rewards left after the decrease
+                var nodeRewardAmount = nodeRecord.tokensStaked.balance * totalRewardScale
+                var nodeRewardsLeft = nodeRewardAmount * rewardDecreasePercentage
+
+                // Add the remaining to the total number of rewards withheld
+                sumRewardsWithheld = sumRewardsWithheld + (nodeRewardAmount - nodeRewardsLeft)
```

Comment:
⚠️  This is the reverse of what I would expect, based on the comments and variable naming. 

`rewardDecreasePercentage` seems intended to mean the percentage of a node's rewards that will be withheld. For example, `0.2` would result in the node getting 80 tokens and 20 tokens being withheld.

`nodeRewardsLeft` is the reward value the node receives after the withholding. But we are computing it like `nodeRewardAmount * rewardDecreasePercentage`. Which means if we say we want to decrease node A's reward by 20%, we will actually decrease it by 80% (and only pay them 20%).

Reply:
thats a good point that some might interpret it that way. That isn't what I intended, but I'll try to think of better naming

Reply:
I'm changing it to `rewardDecreaseToPercentage`